### PR TITLE
Maid 1546

### PR DESCRIPTION
**Pull request by brian-js**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/892)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @canndrew

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/messages.rs [hunk 1]
```diff
@@ -40,7 +41,7 @@ pub enum Message {
 /// Messages sent via a direct connection.
 ///
 /// Allows routing to directly send specific messages between nodes.
-#[derive(Debug, RustcEncodable, RustcDecodable)]
+#[derive(RustcEncodable, RustcDecodable)]
```
Comment (afck) [inline src/messages.rs]: Would it make sense to create some wrapper type `BinaryData(Vec<u8>)` instead, manually implement `Debug` for that one, and then keep the `#[derive(Debug)]` on all the message types?

Comment (brian-js) [inline src/messages.rs]: Maybe create a task for that if it's desired.

Comment (Fraser999) [inline src/messages.rs]: This was previously discussed in https://github.com/maidsafe/routing/issues/721 which was closed without a decent resolution in my opinion (I would have preferred something similar to what you're suggesting).

File: src/messages.rs [hunk 2]
```diff
@@ -363,3 +364,169 @@ pub enum ResponseContent {
...
+                dst: Authority::ClientManager(name),
+                content: RequestContent::Connect,
+            }
+        );
+        let full_id = FullId::new();
+        let signed_message_result = SignedMessage::new(routing_message.clone(), &full_id);
+
+        assert!(signed_message_result.is_ok());
+
+        let signed_message = signed_message_result.unwrap();
```
Comment (Fraser999) [inline src/messages.rs]: Can you replace the bare `unwrap` with our `unwrap_result` macro please?  There are a few instances in these tests.

File: src/messages.rs [hunk 3]
```diff
@@ -363,3 +364,169 @@ pub enum ResponseContent {
...
+    extern crate rand;
+
+    use super::{HopMessage, SignedMessage, RoutingMessage, RequestMessage, RequestContent};
+    use id::FullId;
+    use authority::Authority;
+    use xor_name::XorName;
+    use sodiumoxide::crypto::sign;
+
+    #[test]
+    fn signed_message_check_integrity() {
```
Comment (Fraser999) [inline src/messages.rs]: Looks like we're not really checking anything but the success cases here.  Could we add checks for failure of integrity if any of the three fields of the signed message is tampered with or corrupted please?

Comment (brian-js) [inline src/messages.rs]: The tests are based on the task description, "The non-trivial functions of these structs need unit tests." Andreas asked for the error case for the signature when corrupted in the PR for MAID-1547, that's been done. Do we want all fields to be altered in various combinations then checked in each test?

File: src/messages.rs [hunk 4]
```diff
@@ -364,6 +365,98 @@ pub enum ResponseContent {
...
+        match *self {
+            DirectMessage::BootstrapIdentify { ref public_id, ref current_quorum_size } =>
+                write!(formatter, "DirectMessage {{ BootstrapIdentify {{ {:?}, {:?} }} }}",
+                    public_id, current_quorum_size),
+            DirectMessage::BootstrapDeny =>
+                write!(formatter, "DirectMessage {{ BootstrapDeny }}"),
+            DirectMessage::ClientIdentify { ref client_restriction, .. } =>
+                write!(formatter, "DirectMessage {{ ClientIdentify {{ .., {:?} }} }}", client_restriction),
+            DirectMessage::NodeIdentify { .. } =>
+                write!(formatter, "DirectMessage {{ NodeIdentify {{ .. }} }}"),
```
Comment (afck) [inline src/messages.rs]: Maybe the `serialised_public_id` should be included for `ClientIdentify` and `NodeIdentify`, using `format_binary_array`.

**Comment by afck** [comment]
In `messages.rs`, a lot of content is currently completely omitted from the debug strings (`..`). Maybe it should in general be included, using `format_binary_array`.

**Comment by brian-js** [comment]
I agree there's a range of possibilities here, although I mainly went with task description at https://maidsafe.atlassian.net/browse/MAID-1546, replacing <binary data> with .. in order to reduce the useful debug output to a minimal, and using format_binary_array for unserialised, unencrypted data, with the exception of nonce arguments which may or may not prove useful during debugging. If we need this changed further could you push the task back and put updated specs in a task comment please, and I'll close this PR for now.

**Comment by afck** [comment]
OK, makes sense.

**Pull request merged**